examples/addVectors.cxx

```diff
@@ -0,0 +1,194 @@
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <iostream>
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/defines.hpp"
+
+#define CUDA_BLOCK_SIZE 256
+
+void checkSolution(int *C, int in_N);
+
+
+//User is reposible for memory management
+int * allocate(RAJA::Index_type size) {
+  int * ptr;
+#if defined(RAJA_ENABLE_CUDA)
+  cudaMallocManaged((void**)&ptr, sizeof(int)*size,cudaMemAttachGlobal);
+#else
+  ptr = new int[size];
+#endif
+  return ptr;
+}
+
+void deallocate(int* &ptr) {
+  if (ptr) {
+#if defined(RAJA_ENABLE_CUDA)
+    cudaFree(ptr);
+#else
+    delete[] ptr;
+#endif
+    ptr = nullptr;
+  }
+}
+
+/*Example 1: Adding Two Vectors
+
+  ----[Details]---------------------
+  Adds two vectors of length N
+
+  -----[RAJA Concepts]---------------
+  1. Introduces the forall loop and basic RAJA policies
+
+  RAJA::forall<RAJA::exec_policy>(RAJA::range_policy,[=](index i)) {
+
+         //body
+
+  });
+  
+  ----[Arguments]--------------
+  [=] Pass by copy
+  [&] Pass by reference
+  RAJA::exec_policy  - specifies how the traversal occurs
+  RAJA::range_policy - provides a list in which the index may iterate on
+ */
+int main(int RAJA_UNUSED_ARG(argc), char** RAJA_UNUSED_ARG(argv[]))
+{
+
+  std::cout<<"Example 1: Adding Two Vectors"<<std::endl;
+
+  const int N = 1000;
+  int *A = allocate(N);
+  int *B = allocate(N);
+  int *C = allocate(N);
+  
+  //Populate vectors
+  for(int i=0; i<N; ++i) {
+    A[i] = i;
+    B[i] = i;
+  }
+
+  //----[Standard C++ Loop]---------------
+  std::cout<<"Standard C++ Loop"<<std::endl;
+  for(int i=0; i<N; ++i) {
+    C[i] = A[i] + B[i];
+  }
+  checkSolution(C,N);
+  std::cout<<"\n"<<std::endl;
+  //======================================
+     
+  
+  /*----[RAJA: Sequential Policy]---------  
+  RAJA::seq_exec -  Executes the loop sequentially
+
+  RAJA::RangeSegment(start,stop) - generates a sequence of numbers
+  on which a RAJA loop may iterate on
```

Comment:
I would just add that it generates a contiguous sequence of numbers by the [start, stop) interval specified.